scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java

```diff
@@ -1840,6 +1841,25 @@ public List<WorkflowExecutionTime> getTopExecutionTimeWorkflows(int numberOfWork
                                                       endDate);
     }
 
+    @Override
+    @ImmediateService
+    public CompletedJobsCount getCompletedJobs(Boolean myJobs, String workflowName, String timeWindow)
+            throws NotConnectedException, PermissionException, InvalidTimeWindowException {
+        UserIdentificationImpl ident = frontendState.checkPermission("getCompletedJobs",
+                                                                     "You don't have permissions to get top execution time workflows");
+        String tenant = null;
+        if (PASchedulerProperties.SCHEDULER_TENANT_FILTER.getValueAsBoolean() && !ident.isAllTenantPermission()) {
+            // overwrite tenant filter if the user only has access to his own tenant
+            tenant = ident.getTenant();
+        }
+        TimeWindow timeWindowEnum = TimeWindow.getTimeWindow(timeWindow);
+        if (timeWindowEnum == null) {
+            throw new InvalidTimeWindowException(timeWindow + " is not a valid time window. Accepted values are " +
+                                                 Arrays.toString(TimeWindow.values()));
+        }
+        return dbManager.getCompletedJobs(myJobs ? ident.getUsername() : null, tenant, workflowName, timeWindowEnum);
```

Comment:
are you reading from the database all completed jobs in the given time window ?

Reply:
yes